src/android/CodePush.java

```diff
@@ -436,7 +442,9 @@ public void onStart() {
             /* The application was resumed from the background. */
             /* Handle ON_NEXT_RESUME pending installations. */
             InstallOptions pendingInstall = this.codePushPackageManager.getPendingInstall();
-            if ((pendingInstall != null) && (InstallMode.ON_NEXT_RESUME.equals(pendingInstall.installMode))) {
+            if (pendingInstall != null && 
+                InstallMode.ON_NEXT_RESUME.equals(pendingInstall.installMode) && 
+                (new Date().getTime() - lastPausedTimeMs) / 1000 >= pendingInstall.minimumBackgroundDuration) {
```

Comment:
I would agree if we used this logic multiple times, but it's only here once and on the iOS native side once. I can calculate the difference outside of the if statement if that makes it cleaner (`int backgroundDuration = (new Date().getTime() - lastPausedTimeMs);`) but I don't think an entire function is necessary.